File: CDSS-backend/src/main/java/sbnz/cdss/service/UserService.java
package sbnz.cdss.service;

import org.springframework.stereotype.Service;
import sbnz.cdss.model.entity.User;

import java.util.List;

@Service
public interface UserService {
    User login(String username, String password);
    boolean addDoctor(User doctor);
    List<User> getAllDoctors();

    User addAdministrator(User user);
    User updateUser(User user);
    void deleteUser(User user);
    void deleteUser(Long id);
}

File: CDSS-frontend/src/app/service/user.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient) {
  }


  login(username: string, password: string) {
    return this.http.post('/api/users/login', {'username': username, 'password': password});
  }

  logout() {
    return this.http.post('/api/user/logout', null);
  }

  addDoctor(user) {
    return this.http.post('/api/users/add-doctor', user,{responseType: 'text'});
  }

  getAllDoctors() {
    return this.http.get('/api/users/doctors');
  }
}
